scripts/zones/Kazham/npcs/Magriffon.lua

```diff
@@ -12,14 +17,29 @@ require("scripts/zones/Kazham/TextIDs");
 -----------------------------------
 
 function onTrade(player,npc,trade)
-end; 
+    if(player:getQuestStatus(OUTLANDS, GULLIBLES_TRAVELS) == 1) then
+        tradedGil = trade:getGil();
+        if(tradedGil >= player:getVar("MAGRIFFON_GIL_REQUEST")) then
+            player:startEvent(0x0092);
+        end
+    end
+end;
 
 -----------------------------------
 -- onTrigger Action
 -----------------------------------
 
 function onTrigger(player,npc)
-	player:startEvent(0x008F);
+    if(player:getQuestStatus(OUTLANDS, GULLIBLES_TRAVELS) == 0 and player:getFameLevel(KAZHAM) >= 6) then --Checks if Gullible's Travels hasn't been started or completed and that player has at least 6 Kazham fame
+        player:setVar("MAGRIFFON_GIL_REQUEST", math.random(10000,30000));  --Generates an amount between 20,000 and 30,000 to request from the player
+        player:startEvent(0x0090, 0, player:getVar("MAGRIFFON_GIL_REQUEST"));

```

Comment:
also, since player:getVar("MAGRIFFON_GIL_REQUEST") is used a few times, define it as a local at the top of onTrigger
do the same with the gil request eg
```
local gil = math.random(10000,30000);
local gilRequest = player:getVar("MAGRIFFON_GIL_REQUEST");
```